src/Controller/SearchController.php

```diff
@@ -238,6 +251,89 @@ public function resultsAction(
         ]);
         $filterForm->handleRequest($request);
 
+        if ($filterForm->isSubmitted() && $filterForm->isValid()) {
+
+            $clickedButton = $filterForm->getClickedButton();
+            $buttonName = $clickedButton ? $clickedButton->getName() : '';
+
+            $savedSearch = $searchData->getSelectedSavedSearch();
+
+            // NOTE: if a saved search was selected from the "Manage my views" dropdown, this performs a click (via an
+            // `onchange` attribute) on the form's hidden "load" button; opposed to this, `$buttonName` will be empty
+            // if the search params get changed for an existing saved search via the "Restrict results" form part
+            if ($buttonName === 'load' && $savedSearch) {
+                $savedSearchURL = $savedSearch->getSearchUrl();
+
+                if ($savedSearchURL) {
+                    // redirect to the search_url stored for the chosen saved search
+                    $redirectResponse = new RedirectResponse($request->getSchemeAndHttpHost() . $savedSearchURL);
+
+                    return $redirectResponse;
+                }
+
+            } elseif ($buttonName === 'delete' && $savedSearch) {
+                $savedSearchesService->removeSavedSearch($savedSearch);
+
+                // remove the "delete" param as well as saved search related params from current search URL
+                $request = $this->setSubParamForRequestQueryParam('delete', null, 'search_filter', $request);
+                $request = $this->setSubParamForRequestQueryParam('selectedSavedSearch', null, 'search_filter', $request);
+                $request = $this->setSubParamForRequestQueryParam('selectedSavedSearchTitle', null, 'search_filter', $request);
+                $searchURL = $this->getUpdatedRequestUriForRequest($request);
+
+                $redirectResponse = new RedirectResponse($request->getSchemeAndHttpHost() . $searchURL);
+
+                return $redirectResponse;
```

Comment:
Hier wäre redirectToRoute(...) schöner.

```suggestion
                return $this->redirect($request->getSchemeAndHttpHost() . $searchURL);
```